### PR TITLE
Improve BQ PDAO test for full views.

### PR DESCRIPTION
**Pull request by danxmoran**
While debugging #525 I found that the full-view test in the BQ PDAO suite wasn't fully cleaning up after itself. Digging into it a bit more, I saw that it wasn't really asserting on anything... so I beefed it up a little

### CODE REVIEW

File: src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java [hunk 1]
```diff
@@ -403,17 +405,65 @@ private void softDeleteRows(BigQueryProject bq,
...
-        DatasetTable fileTable = getTable(dataset, "file");
-        List<DatasetTable> tables = Arrays.asList(participantTable, sampleTable, fileTable);
+            // Create a full-view snapshot!
+            DatasetSummaryModel datasetSummary = DatasetJsonConversion.datasetSummaryModelFromDatasetSummary(
+                dataset.getDatasetSummary());
+            SnapshotSummaryModel snapshotSummary = connectedOperations.createSnapshot(datasetSummary,
+                "snapshot-fullviews-test-snapshot.json", "");
+            SnapshotModel snapshot = connectedOperations.getSnapshot(snapshotSummary.getId());
 
-        bigQueryPdao.createSnaphotTableFromLiveViews(bigQueryProject, tables, datasetBqDatasetName);
```
Comment (danxmoran) [inline src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java:416]: This method was returning a `String` (the result of generating SQL templates), not creating the BQ resources

File: src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java [hunk 2]
```diff
@@ -403,17 +405,65 @@ private void softDeleteRows(BigQueryProject bq,
...
+
+            Assert.assertThat(participantIds, containsInAnyOrder(
+                "participant_1", "participant_2", "participant_3", "participant_4", "participant_5"));
+            Assert.assertThat(sampleIds, containsInAnyOrder(
+                "sample1", "sample2", "sample3", "sample4", "sample5", "sample6", "sample7"));
+            Assert.assertThat(fileIds, is(equalTo(Collections.singletonList("file1"))));
+        } finally {
+            storage.delete(participantBlob.getBlobId(), sampleBlob.getBlobId(), fileBlob.getBlobId());
+            bigQueryPdao.deleteDataset(dataset);
+            datasetDao.delete(dataset.getId());
```
Comment (danxmoran) [inline src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java:476]: This is the extra cleanup logic I originally wanted to add. Without it, the `tearDown` method tries to delete the resource profile & encounters a FK integrity violation

Comment (RoriCremer) [reply src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java:476]: maybe add this as a comment in the code? I'm curious as to how we were getting around the FK integrity violation and whether we could get into the same state in other tests.

**Review by RoriCremer — APPROVED** [review]

**Review by fboulnois — APPROVED** [review]
The revised test lgtm. The FK violation is worrying.

File: src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java (hunk 2 above)
Comment (danxmoran) [reply src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java:476]: Happy to add a comment somewhere, but not sure of the best place (this diff makes the test case look the same as all the other tests in the file). I could add the comment to the `tearDown` line that triggers the FK violation, but I'm not sure who would read it... Alt I could add it to every test case for max copy-paste-ability. WDYT?

Comment (danxmoran) [reply src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoTest.java:476]: Decided to do both, more comments don't hurt anybody

**Pull request merged**